pkg/relayer/relayminer.go

```diff
@@ -134,3 +136,48 @@ func (rel *relayMiner) ServePprof(ctx context.Context, addr string) error {
 
 	return nil
 }
+
+// ServePing exposes ping HTTP server to check the reachability between the
+// relay miner and its dependencies (Ex: relay server and their respective
+// backend URLs).
+func (rel *relayMiner) ServePing(ctx context.Context, network, addr string) error {
+	ln, err := net.Listen(network, addr)
+	if err != nil {
+		return err
+	}
+
+	// Start a long-lived goroutine that starts an HTTP server responding to
+	// ping requests. A single ping request on the relay server broadcasts a
+	// ping to all backing services/data nodes.
+	go func() {
+		if err := http.Serve(ln, rel.newPinghandlerFn(ctx)); err != nil && !errors.Is(http.ErrServerClosed, err) {
+			rel.logger.Error().Err(err).Msg("ping server unexpectedly closed")
+		}
+	}()
+
+	go func() {
+		<-ctx.Done()
```

Comment:
#PUC when we expect this context to close.

I'm guessing (assumption, gut intuition) is when the process shuts down, but making it explicit would be nice.

Reply:
Added here 👍🏾  f1faa926652466aebe7a5c04db76a8a1133f0311